src/pretix/plugins/socialauth/mediawiki_provider.py

```diff
@@ -0,0 +1,174 @@
+    """
+    User-friendly exception for MediaWiki rate limiting.
+    
+    This exception provides vague time descriptions instead of exact retry times
+    to avoid rushing users while still giving helpful guidance.
+    """
+    
+    def __init__(self, retry_after_seconds=None, message=None):
+        self.retry_after_seconds = retry_after_seconds
+        
+        if message:
+            super().__init__(message)
+        else:
+            friendly_time = self._get_friendly_time_description(retry_after_seconds)
+            super().__init__(
+                _("MediaWiki is currently experiencing high traffic. "
+                  "Please try again in {time}.").format(time=friendly_time)
+            )
+    
+    def _get_friendly_time_description(self, seconds):
+        """
+        Convert exact seconds to user-friendly vague descriptions.
+        
+        This ensures users get helpful guidance without feeling rushed
+        by exact countdown timers.
+        """
+        if not seconds or seconds <= 0:
+            return _("a moment")
+        elif seconds <= 30:
+            return _("a minute")
+        elif seconds <= 120:
+            return _("a couple of minutes") 
+        elif seconds <= 300:
+            return _("a few minutes")
+        elif seconds <= 600:
+            return _("several minutes")
+        else:
+            return _("a little while")
+
+
+class RateLimitedOAuth2Client:
+    """
+    OAuth2 client wrapper that handles rate limiting gracefully.
+    
+    Wraps the standard OAuth2 client to detect 429 responses and convert
+    them to user-friendly exceptions without retry logic.
+    """
+    
+    def __init__(self, client):
+        self.client = client
+    
+    def get_access_token(self, code, pkce_code_verifier=None):
+        """
+        Get access token with rate limit handling.
+        
+        Makes the OAuth2 request and converts any 429 responses to
+        user-friendly MediaWikiRateLimitException.
+        """
+        try:
+            return self.client.get_access_token(code, pkce_code_verifier)
+        except OAuth2Error as e:
+            if hasattr(e, 'response') and e.response.status_code == 429:
+                # Extract retry-after header if present
+                retry_after = None
+                if 'Retry-After' in e.response.headers:
+                    try:
+                        retry_after = int(e.response.headers['Retry-After'])
+                    except (ValueError, TypeError):
+                        retry_after = None
+                
+                logger.warning(
+                    "MediaWiki OAuth rate limit hit, returning user-friendly error. "
+                    "Original retry-after: %s seconds", 
+                    retry_after
+                )
+                
+                raise MediaWikiRateLimitException(retry_after)
+            raise
+
+
+class MediaWikiAccount(ProviderAccount):
```

Comment:
I wonder why do you define `MediaWikiAccount`? Doesn't it already exist in `django-allauth`?